probe/tag/tagger.go

```diff
@@ -0,0 +1,25 @@
+package tag
+
+import "github.com/weaveworks/scope/report"
+
+// Tagger tags nodes with value-add node metadata.
+type Tagger interface {
+	Tag(r report.Report, ts report.TopologySelector, id string) report.NodeMetadata
```

Comment:
@peterbourgon Whats the thoughts behind this interface?  Its strikes me as overly general and not particularly useful.  For one, we only have two implementations, and one call site - did you say we need three of something before we think about unification?

Also, the docker_tagger is going to become much more than just a tagger (its going to have to emit at least one topology itself).

Finally, the interface (report, selector, id) seems weird.  It seems a bit tortuous, and is only ever called in a loop by tagger. 

I suggest for topology tagger, we move the loop into the tagger.   For docker tagger, we make it take the process topology explicitly.  And we get rid of this interface.